plugins/tactics/src/Ide/Plugin/Tactic.hs

```diff
@@ -226,20 +251,26 @@ judgementForHole state nfp range = runMaybeT $ do
   resulting_range <- liftMaybe $ toCurrentRange amapping $ realSrcSpanToRange rss
 
   let hyps = hypothesisFromBindings rss binds
-  pure (amapping, resulting_range, Judgement hyps $ CType goal)
+  pure (amapping, b2, resulting_range, mkFirstJudgement hyps goal, dflags)
+
 
 
 tacticCmd :: (OccName -> TacticsM ()) -> CommandFunction TacticParams
 tacticCmd tac lf state (TacticParams uri range var_name)
   | Just nfp <- uriToNormalizedFilePath $ toNormalizedUri uri =
       fromMaybeT (Right Null, Nothing) $ do
-        (pos, _, jdg) <- MaybeT $ judgementForHole state nfp range
-        -- Ok to use the stale 'ModIface', since all we need is its 'DynFlags'
-        -- which don't change very often.
-        (hscenv, _) <- MaybeT $ runIde state $ useWithStale GhcSession nfp
-        let dflags = hsc_dflags $ hscEnv hscenv
+        (pos, b2, _, jdg, dflags) <- MaybeT $ judgementForHole state nfp range
+        range' <- liftMaybe $ toCurrentRange pos range
+        (tcmod, _) <- MaybeT $ runIde state $ useWithStale TypeCheck nfp
+        let span = rangeToRealSrcSpan (fromNormalizedFilePath nfp) range'
+            -- TODO(sandy): unclear if this span is correct; might be
+            -- pointing to the wrong version of the file
```

Comment:
fix this TODO by applying the position mapping returned by `useWithStale`

Reply:
done